File: pinyougou-Parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;
import entity.Result;

import java.util.List;
import java.util.Map;

/**
 * 品牌接口
 * @author Youla
 *
 */
public interface BrandService {

		public List<TbBrand> findAll();

		public PageResult findByPage(int page, int size);

		public Result add(TbBrand tbBrand);

		public TbBrand findById(long id);

		public Result update(TbBrand tbBrand);

		public Result delete(long[] ids);

		public PageResult searchByExample(TbBrand tbBrand,int page,int size);

		public List<Map> findBrandToMap();
}

File: pinyougou-Parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//服务层
app.service('brandService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../brand/findAll.do');		
	}
	//分页 
	this.findPage=function(page,size){
		return $http.get('../brand/findByPage.do?page='+page+'&size='+size);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../brand/findById.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../brand/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../brand/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../brand/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,size,searchEntity){
		return $http.post('../brand/search.do?page='+page+"&size="+size, searchEntity);
	}

	//将数据封装为Map集合 再转换为json字符串  返回

	this.brandMap=function(){
		return $http.get('../brand/findBrandToMap.do');
	}

});
